c_src/sqlite3_nif.c

```diff
@@ -1065,6 +1065,56 @@ exqlite_set_update_hook(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
     return make_atom(env, "ok");
 }
 
+//
+// Log Notifications
+//
+
+ErlNifPid* log_hook_pid = NULL;
```

Comment:
There can only be one, hence a global variable. Maybe we can add a mutex too.

Reply:
Regarding the mutex, I think the use-case would be to set it on app startup once, so the mutex there is not necessary.

```elixir
defmodule Some.Application do
  use Application
  def start(_, _) do
     :ok  = Exqlite.Sqlite3.set_log_hook(some_logger_pid)
     # etc.
   end
end
```

Reply:
Definitely need a mutex here. https://www.erlang.org/doc/man/erl_nif#enif_mutex_create

Reply:
Also can you lift the global to be near the top of the file?

Reply:
done in https://github.com/elixir-sqlite/exqlite/pull/266/commits/919d2ef91fc3ad78ad583ab68d0c11686fe3239d